### PR TITLE
new err messages

### PR DESCRIPTION
**Pull request by marcin-mogiela**

### CODE REVIEW

File: app/routes/govuk-components/validate-date-input-year.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            is: Joi.number().required(),
+            then: Joi.when(`${namePrefix}-month`, {
+              is: Joi.number().required(),
+              then: Joi.custom(customValidation)
+            })
+          })
+      }
+    ]
+  }).messages({
+    'string.length': 'Year must include 4 numbers',
```
Comment (chrissmKainos) [inline app/routes/govuk-components/validate-date-input-year.js:36]: No check on 'number.base', 'number.min', 'number.max' as done for the other date property validations, is there a reason for this?

File: app/routes/govuk-components/validate-date-input-day.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+const Joi = require('joi')
+
+const isYearEmpty = (helpers, namePrefix) => helpers.state.ancestors[0][`${namePrefix}-year`] === ''
+const isMonthEmpty = (helpers, namePrefix) => helpers.state.ancestors[0][`${namePrefix}-month`] === ''
+const isLeapYear = (year) => (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0
+
+const validateDateInputDay = (namePrefix, whateverItIs) => {
```
Comment (chrissmKainos) [inline app/routes/govuk-components/validate-date-input-day.js:7]: could you possibly change the property named 'whateverItIs' to something a bit more descriptive (similar comment for all validation files)

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ffc-ahwr-farmer-claim",
-  "version": "0.35.1",
+  "version": "0.35.2",
```
Comment (chrissmKainos) [inline package.json:3]: Probably should be a minor bump up to 0.36.0

File: app/routes/govuk-components/validate-date-input-day.js (hunk 2 above)
Comment (marcin-mogiela) [reply app/routes/govuk-components/validate-date-input-day.js:7]: image

File: app/routes/govuk-components/validate-date-input-day.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            }
+          ]
+        }).required()
+      }
+    ]
+  }).messages({
+    'number.base': `${whateverItIs} must be a real date`,
+    'number.min': `${whateverItIs} must be a real date`,
+    'number.max': `${whateverItIs} must be a real date`,
+    'dateInputDay.ifNothingIsEntered': `Enter the ${whateverItIs.toLowerCase()}`,
```
[the remark this answers is not in the document]
Comment (marcin-mogiela) [reply app/routes/govuk-components/validate-date-input-day.js]: done

File: app/routes/govuk-components/validate-date-input-year.js (hunk 1 above)
Comment (marcin-mogiela) [reply app/routes/govuk-components/validate-date-input-year.js]: done

File: app/routes/govuk-components/validate-date-input-day.js (hunk 2 above)
Comment (marcin-mogiela) [reply app/routes/govuk-components/validate-date-input-day.js]: renamed to dateName

File: package.json (hunk 3 above)
Comment (marcin-mogiela) [reply package.json]: bumped

**Review by chrissmKainos — APPROVED** [review]

**Pull request merged**